src/server/models/Reading.js

```diff
@@ -245,158 +207,57 @@ class Reading {
-	 */
-	static async getCompressedReadings(meterIDs, fromTimestamp = null, toTimestamp = null, numPoints = 500, conn) {
-		const allCompressedReadings = await conn.func('compressed_readings',
-			[meterIDs, fromTimestamp || '-infinity', toTimestamp || 'infinity', numPoints]);
-		// Separate the result rows by meter_id and return a nested object.
-		const compressedReadingsByMeterID = mapToObject(meterIDs, () => []); // Returns { 1: [], 2: [], ... }
-		// For each row in the allCompressedReadings table, append the compressed reading value to the array for
-		// the meter that corresponds to that reading.
-		for (const row of allCompressedReadings) {
-			compressedReadingsByMeterID[row.meter_id].push(
-				{ reading_rate: row.reading_rate, start_timestamp: row.start_timestamp, end_timestamp: row.end_timestamp }
-			);
-		}
-		return compressedReadingsByMeterID;
-	}
-
-	/**
-	 * Gets a number of compressed readings that approximate the given time range for the given groups.
-	 *
-	 * Compressed readings are in kilowatts.
-	 * @param groupIDs an array of ids for groups whose points are being compressed
-	 * @param {Moment?} fromTimestamp An optional start point for the time range.
-	 * @param {Moment?} toTimestamp An optional end point for the time range
-	 * @param numPoints The number of points to compress to. Defaults to 500
-	 * @param conn is the connection to use.
-	 * @return {Promise<object<int, array<{reading_rate: number, start_timestamp: Moment, end_timestamp: Moment}>>>}
-	 */
-	static async getCompressedGroupReadings(groupIDs, fromTimestamp = null, toTimestamp = null, numPoints = 500, conn) {
-		const allCompressedReadings = await conn.func('compressed_group_readings',
-			[groupIDs, fromTimestamp || '-infinity', toTimestamp || 'infinity', numPoints]);
-		// Separate the result rows by meter_id and return a nested object.
-		const compressedReadingsByGroupID = mapToObject(groupIDs, () => []); // Returns { 1: [], 2: [], ... }
-		for (const row of allCompressedReadings) {
-			compressedReadingsByGroupID[row.group_id].push(
-				{ reading_rate: row.reading_rate, start_timestamp: row.start_timestamp, end_timestamp: row.end_timestamp }
-			);
-		}
-		return compressedReadingsByGroupID;
-	}
-
-	/**
-	 * Gets barchart readings for every meter across the given time interval for a duration.
-	 *
-	 * Compressed readings are in kilowatts.
-	 * @param meterIDs an array of ids for meters whose points are being compressed
-	 * @param duration A moment time duration over which to sum the readings
-	 * @param fromTimestamp An optional start point for the beginning of the entire time range.
-	 * @param toTimestamp An optional end point for the end of the entire time range.
-	 * @param conn is the connection to use.
-	 * @return {Promise<object<int, array<{reading_sum: number, start_timestamp: Date, end_timestamp: Date}>>>}
-	 */
-	static async getBarchartReadings(meterIDs, duration, fromTimestamp = null, toTimestamp = null, conn) {
```

Comment:
Could you please explain why this function is removed? I saw that this function is still used in src/server/test/db/barchartReadingsTests.js. Thank you.

Reply:
Yes, you are correct. To move this forward faster, I have not been updating the test code (as noted in the PR). As a result, the test code does not reflect the new functions nor ones that were obsolete. This is going to happen later (hopefully not too long now). In this case, the test code either needs to have the function name change (that also means modifying the parameter list to deal with the new unit argument and testing it) if there is a new equivalent or removing the tests for obsolete code. Thus, it will take some effort to do all the test changes including creating the new tests for new capabilities. So, I felt it was best to clean up the code of obsolete code whose only use was by test code (which indicates it not really used by the project any more but I also did other checks to try to be careful). Doing OED code this way is unusual but acceptable right now since it isn't on the development branch (that should always pass the tests). The plan was this would allow for more rapid integration and use by the other teams working on resource generalization so faster progress in seeing results and finding any fundamental issues in this complex change. It does mean we are acquiring technical debt by not testing and some code may need future fixes. The plan is to pay down that debt soon and before we fully accept the new code as part of OED (as in merging into development). I hope this helps. I agree the motivation was not clear and it is unusual for OED. The design document for this tried to indicate that plan by making the full testing of the code independent of the implementation and somewhat later in the figure.

Reply:
Thank you for your explanation. This makes sense to me.